lib/LaTeXML/Package/amscd.sty.ltxml

```diff
@@ -101,22 +97,34 @@ DefConstructor('\cd@stack Undigested {} ScriptStyle ScriptStyle', sub {
   properties => { scriptpos => sub { "mid" . $_[0]->getScriptLevel; } },
   reversion  => '@#1{#3}#1{#4}#1');
 
-# Temporary...
-# Later deal with vertically centering the side things, parser issues...
-
-DefConstructor('\cd@adjacent Undigested {} ScriptStyle ScriptStyle', sub {
+DefConstructor('\lx@amscd@adjacent Undigested {} ScriptStyle ScriptStyle', sub {
     my ($document, $reversion, $op, $left, $right, %props) = @_;
+    # Make the left & right parts width=0, so they don't affect centering
     $document->openElement('ltx:XMWrap', role => 'ARROW');    # Role?
-    $document->insertElement('ltx:XMArg', $left) unless IsEmpty($left);
+    $document->insertElement('ltx:XMArg', $left, width => Dimension(0)) unless IsEmpty($left);
     $document->insertElement('ltx:XMArg', $op);
-    $document->insertElement('ltx:XMArg', $right) unless IsEmpty($right);
+    $document->insertElement('ltx:XMArg', $right, width => Dimension(0)) unless IsEmpty($right);
     $document->closeElement('ltx:XMWrap'); },
   reversion => '@#1{#3}#1{#4}#1');
 
-# This isn't really having the desired effect when transformed to MathML and
-# displayed in Firefox.... have I got it right; has Firefox???
-DefMath('\leftarrowfill@',  "\x{2190}", role => 'ARROW', stretchy => 'true');
-DefMath('\rightarrowfill@', "\x{2192}", role => 'ARROW', stretchy => 'true');
+# These, in case used...
+DefMacro('\leftarrowfill@ {}',      '\lx@amscd@leftarrow');
+DefMacro('\rightarrowfill@ {}',     '\lx@amscd@rightarrow');
+DefMacro('\leftrightarrowfill@ {}', '\lx@amscd@leftrightarrow');
+
+# These are stretchy, widenned version; should be \minCDarrowsidth or \minaw@ wide
+DefPrimitive('\lx@amscd@leftarrow', sub {
+    Box("\x{2190}", undef, undef, '\leftarrow',
+      role => 'ARROW', stretchy => 'true', meaning => 'rightarrow', width => Dimension('30pt')); });
+DefPrimitive('\lx@amscd@rightarrow', sub {
+    Box("\x{2192}", undef, undef, '\rightarrowf',
+      role => 'ARROW', stretchy => 'true', meaning => 'rightarrow', width => Dimension('30pt')); });
+DefPrimitive('\lx@amscd@leftrightarrow', sub {
+    Box("\x{2194}", undef, undef, '\leftrightarrow',
+      role => 'ARROW', stretchy => 'true', meaning => 'leftrightarrow', width => Dimension('30pt')); });
+DefPrimitive('\lx@amscd@equals', sub {
+    Box("=", undef, undef, T_OTHER('='),
+      role => 'ARROW', stretchy => 'true', meaning => 'equals', width => Dimension('30pt')); });
```

Comment:
As a clarifying question:

What is the intended meaning of an `XMTok` that has a set positive width (30pt here), with `stretchy="true"` at the same time? 

Does the XMath `width` act as `min-width` in this circumstance?